tests/lit-tests/707.ispc

```diff
@@ -0,0 +1,58 @@
+// This test checks that the compiler supports `signed`, `unsigned` and `const`
+// quialifiers in `new` expressions and template arguments. 
+// It also check the compiler does not crash (fatal error).
+
+// RUN: %{ispc} --target=host --emit-llvm-text --nostdlib --nowrap -g %s -o - 2>&1 | FileCheck %s
+
+// CHECK-NOT: FATAL ERROR:
+
+uniform uint* uniform test_uint() {
+    return uniform new uniform unsigned int;
+}
+
+/*
+// Do not work yet due to #3472 waiting to be merged
+uniform int* uniform test_sint() {
+    return uniform new uniform signed int; 
+}
+*/
+
+uniform const int* uniform test_const() {
+    return uniform new uniform const int(815);
+}
+
+const unsigned uniform int* uniform test_mixed_unordered() {
+    return uniform new unsigned const uniform int(815);
+}
+
+/*uniform int<4>* uniform test_vec() {
+    return uniform new uniform int<4>; // Should be legit (could work so far) if short-vectors would be supported.
+}
+
+uniform int<6>* uniform test_vec_with_init1() {
+    uniform int<6> v = {4, 8, 15, 16, 23, 42};
+    return uniform new uniform int<6>(v); // Should be legit (does not work yet) if short-vectors would be supported.
+}
+
+uniform int<6>* uniform test_vec_with_init2() {
+    return uniform new uniform int<6>(4, 8, 15, 16, 23, 42); // Is is legit? (could build without apparent issue so far)
+}
+
+uniform int<6>* uniform test_vec_with_init3() {
+    return uniform new uniform int<6>({4, 8, 15, 16, 23, 42}); // Is is legit? (does not build yet)
```

Comment:
This seems to me more like advanced feature of C++. It can be supported in ISPC, but it's optional.
Other cases looks legit to me.

Reply:
Does your comment reference the last example/function or the whole (commented) block about short vectors?

Reply:
About the last example. Everything else looks reasonable.